### DISCUSSION TITLE
Cloud/Internet required?

### DISCUSSION THREAD

**Discussion by crpeck** [sebr/bhyve-home-assistant#124] [General]
Is access to the Cloud, aka BeeHive servers, required for this to work, or, are commands sent locally? I need a new timer & this looks perfect. I especially like the ability to base waterings on the weather.

**Comment by sebr** [comment]
Yes, all interaction with the devices is done via BHyve's servers. Unfortunately there is no option for local control.

**Reply by crpeck** [reply]
Thanks for the reply, that is unfortunate, but, understandable.

**Reply by GrahamCobb** [reply]
I'm on the same search! If you find one which works without the cloud maybe @sebr wouldn't mind if you mentioned it here?

**Reply by pcsmith811** [reply]
> Yes, all interaction with the devices is done via BHyve's servers. Unfortunately there is no option for local control.

Is this statement still true with HA's addition of a bluetooth integration?

**Reply by sebr** [reply]
It may be possible to reverse engineer the Bluetooth communication, however this is not on my personal todo list. If someone wishes to investigate or even submit a PR I would be very happy to consider for inclusion.

**Reply by vahaldor** [reply]
I just came back from holidays and my garden is almost dead. BHyves stopped opening on 1st September and I it looks it is related to the update fixing a bug on this date. Was this caused by the b-hyve cloud? If so, you may consider developing only the BT link to let us open the valve if the cloud has some issues. I am using the common B-Hyve hose tap and control only its opening.

**Reply by harrv** [reply]
> Yes, all interaction with the devices is done via BHyve's servers. Unfortunately there is no option for local control.

Thank you for that answer. It took me a while to find it. I think it would be great to mention this in the README (both here and in HACS), or if it is, and I somehow missed it, to make it more prominent. I'm shopping for an irrigation controller, want to integrate it with Home Assistant, and local control is important to me. I doubt I'm unique.

**Reply by dxmnkd316** [reply]
I believe Eve has a full-local device but it's quite spendy.  Otherwise I've seen several DIY cheapo Ali Express solenoids with an ESP controller.  Not sure I'd trust those with the potential consequences of a fail-open water line on the yard.
